### PR TITLE
M3 point bundle

### PR DESCRIPTION
**Pull request by Dcoutelle**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/7997
| BC breaks? | 
| Deprecations? | 

Description:
This PR upgrades the Symfony Form types to work with Symfony 3.
This is specific to the Components > Point & trigger Content
Things to test:

Creating point en trigger, run them

### CODE REVIEW

**Comment by npracht** [comment]
I cannot create a point action
Capture d’écran 2019-11-01 à 13 16 44
```
Uncaught PHP Exception Symfony\Component\Form\Exception\InvalidArgumentException: "Could not load type "pointaction": class does not exist." at /var/app/current/code/pulls/8052/vendor/symfony/form/FormRegistry.php line 88
```

**Comment by Dcoutelle** [comment]
Trigger action based on points sum is link to https://github.com/mautic/mautic/issues/8066

**Comment by Dcoutelle** [comment]
@escopecz it's done!

**Comment by escopecz** [comment]
@Dcoutelle thanks! Travis is complaining. Please take a look

File: app/bundles/PointBundle/Form/Type/PointType.php [hunk 1]
```diff
@@ -33,17 +39,20 @@ class PointType extends AbstractType
...
+     * @var \Symfony\Component\Translation\TranslatorInterface
      */
     private $translator;
 
     /**
-     * @param MauticFactory $factory
+     * PointType constructor.
+     *
+     * @param \Symfony\Component\Translation\TranslatorInterface      $translator
+     * @param \Mautic\CoreBundle\Security\Permissions\CorePermissions $security
```
Comment (escopecz) [inline app/bundles/PointBundle/Form/Type/PointType.php]:
```suggestion
     * @param CorePermissions $security
```
Use relative path as the classes are defined in the use statements above.

File: app/bundles/PointBundle/Form/Type/TriggerType.php [hunk 2]
```diff
@@ -145,46 +153,48 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-                'label_attr' => ['class' => 'control-label'],
-                'attr'       => [
-                    'class'       => 'form-control',
-                    'data-toggle' => 'datetime',
-                ],
-                'format'   => 'yyyy-MM-dd HH:mm',
-                'required' => false,
-            ]
+          'publishUp',
+          'datetime',
```
Comment (escopecz) [inline app/bundles/PointBundle/Form/Type/TriggerType.php]:
```suggestion
          DateTimeType::class,
```

File: app/bundles/PointBundle/Model/PointModel.php [hunk 3]
```diff
@@ -254,7 +248,7 @@ public function triggerAction($type, $eventDetails = null, $typeId = null, Lead
                     'points'     => $action->getDelta(),
                 ],
                 'lead'         => $lead,
-                'factory'      => $this->factory, // WHAT?
+                'factory'      => '', // WHAT?
```
Comment (escopecz) [inline app/bundles/PointBundle/Model/PointModel.php]: Isn't it used somewhere? It's not that easy to get rid of MauticFactory 😆 

If it the work is out of scope for this issue, please create a new issue instead so we won't forget to refactor it properly.

File: app/bundles/PointBundle/Model/TriggerModel.php [hunk 4]
```diff
@@ -373,7 +367,7 @@ private function invokeCallback($event, Lead $lead, array $settings)
         $args = [
           'event'   => $event,
           'lead'    => $lead,
-          'factory' => $this->factory, // WHAT??
+          'factory' => '',
```
Comment (escopecz) [inline app/bundles/PointBundle/Model/TriggerModel.php]: Same here.

**Comment by escopecz** [comment]
Closing in favour of https://github.com/mautic/mautic/pull/8230. I was not able to push to your branch.

**Pull request closed without merging**